app/javascript/components/timeline-options/helper.js

```diff
@@ -0,0 +1,50 @@
+const smartOrStatements = (group, array) => {
+  let shortUrl = '';
+  let index = array.length - 1;
+  while (index > 0) {
+    shortUrl += `&filter[]=or%20${group}=${array[index]}`
+    index -= 1;
+  }
+  shortUrl = `&filter[]=${group}=${array[0]}` + shortUrl;
+  return shortUrl;
+}
+
+/** Function to build the URL to populate the timeline data table. */
+export const buildUrl = (values) => {
+  var url = `/api/event_streams?limit=5000&offset=0&expand=resources&attributes=group,group_level,group_name,id,event_type,message,ems_id,type,timestamp,created_on`
```

Comment:
Note, this API could easily return the 5000 limit.  We should make sure these pages don't take minutes to load if there's 5000 matching events.  We might have to use paging with offset and limits.

Reply:
yes, since pagination was not implemented in the old table, we can try to introduce them in a different PR. Also, `MiqDataTable` component which uses the pagination feature was reused from the existing API

Reply:
Was the old table limited to 5000 records also?

Reply:
Note that I don't think that 5000 number will be honored anyway - the API has a hard max of 1000: https://github.com/ManageIQ/manageiq-api/blob/master/config/settings.yml#L4.  This is to prevent someone from blowing out the database.